File: csharp/Textures.cs
using Microsoft.Xna.Framework.Graphics;
using StardewModdingAPI;
using StardewValley;

namespace DontStarve;

internal static class Textures {
	public static Texture2D sanityContainer = null!;
	public static Texture2D hungerContainer = null!;
	private static Texture2D backingSanityFiller = null!;
	private static Texture2D backingHungerFiller = null!;

	internal static Texture2D sanityFiller {
		get {
			var color = BarsInformation.sanityColor;
			backingSanityFiller.SetData(new[] { color });
			return backingSanityFiller;
		}
	}

	internal static Texture2D hungerFiller {
		get {
			var color = BarsInformation.hungerColor;
			backingSanityFiller.SetData(new[] { color });
			return backingSanityFiller;
		}
	}

	internal static void loadTextures(IModContentHelper modContent) {
		sanityContainer = modContent.Load<Texture2D>("assets/sanity/bar.png");
		backingSanityFiller = new Texture2D(Game1.graphics.GraphicsDevice, 1, 1);
		hungerContainer = modContent.Load<Texture2D>("assets/hunger/bar.png");
		backingHungerFiller = new Texture2D(Game1.graphics.GraphicsDevice, 1, 1);
	}
}

File: kotlin/src/nativeMain/kotlin/Textures.kt
import wrapper.Game1
import wrapper.ModContentHelper
import wrapper.Texture2D

object Textures {
	lateinit var sanityContainer: Texture2D
	lateinit var hungerContainer: Texture2D
	private lateinit var backingSanityFiller: Texture2D
	private lateinit var backingHungerFiller: Texture2D

	val sanityFiller: Texture2D
		get() {
			val color = BarsInformation.sanityColor
			backingSanityFiller.setData(color)
			return backingSanityFiller
		}

	val hungerFiller: Texture2D
		get() {
			val color = BarsInformation.hungerColor
			backingSanityFiller.setData(color)
			return backingSanityFiller
		}

	fun loadTextures(modContent: ModContentHelper) {
		sanityContainer = modContent.load("assets/sanity/bar.png")
		backingSanityFiller = Texture2D(Game1.graphics.GraphicsDevice, 1, 1)
		hungerContainer = modContent.load("assets/hunger/bar.png")
		backingHungerFiller = Texture2D(Game1.graphics.GraphicsDevice, 1, 1)
	}
}